compiler-rt/test/asan/TestCases/Linux/initialization-nobug-lld.cpp

```diff
@@ -0,0 +1,14 @@
+// RUN: %clangxx_asan -O3 %S/../initialization-nobug.cpp %S/../Helpers/initialization-nobug-extra.cpp -fuse-ld=lld -o %t && %env_asan_opts=check_initialization_order=true:report_globals=3 %run %t 2>&1 | FileCheck %s --implicit-check-not "DynInit"
+
+// Same as initialization-nobug.cpp, but with lld we expect just one
+// `DynInitUnpoison` executed after `AfterDynamicInit` at the end.
+// REQUIRES: lld-available
+
+// With dynamic runtimes `AfterDynamicInit` will called before `executable`
+// contructors, with constructors of dynamic runtime.
+// XFAIL: asan-dynamic-runtime
```

Comment:
UNSUPPORTED is more suitable as this is a fundamental constraint.

Reply:
I don't think XFAIL has anything to do with `fundamental`

It should be appropriate to use where ever the test is consistently fail, like here.